xbmc/video/guilib/VideoStreamSelectHelper.cpp

```diff
@@ -0,0 +1,480 @@
+{
+  bool operator()(const VideoStreamInfoExt& a, const VideoStreamInfoExt& b)
+  {
+    if (a.language != b.language)
+    {
+      return a.language < b.language;
+    }
+    if (a.codecName != b.codecName)
+    {
+      return a.codecName < b.codecName;
+    }
+    if (a.hdrType != b.hdrType)
+    {
+      return a.hdrType < b.hdrType;
+    }
+    if (a.fpsRate != b.fpsRate)
+    {
+      return a.fpsRate < b.fpsRate;
+    }
+    if (a.fpsScale != b.fpsScale)
+    {
+      return a.fpsScale < b.fpsScale;
+    }
+    if (a.height != b.height)
+    {
+      return a.height < b.height;
+    }
+    if (a.width != b.width)
+    {
+      return a.width < b.width;
+    }
+    return a.bitrate < b.bitrate;
+  }
+};
+
+struct SortComparerStreamAudio
+{
+  bool operator()(const AudioStreamInfoExt& a, const AudioStreamInfoExt& b)
+  {
+    if (a.languageDesc != b.languageDesc)
+    {
+      return a.languageDesc < b.languageDesc;
+    }
+    if (a.codecName != b.codecName)
+    {
+      return a.codecName < b.codecName;
+    }
+    if (a.channels != b.channels)
+    {
+      return a.channels < b.channels;
+    }
+    if (a.isDefault != b.isDefault)
+    {
+      return a.isDefault < b.isDefault;
+    }
+    if (a.isForced != b.isForced)
+    {
+      return a.isForced < b.isForced;
+    }
+    if (a.isOriginal != b.isOriginal)
+    {
+      return a.isOriginal < b.isOriginal;
+    }
+    if (a.isHearingImpaired != b.isHearingImpaired)
+    {
+      return a.isHearingImpaired < b.isHearingImpaired;
+    }
+    if (a.isVisualImpaired != b.isVisualImpaired)
+    {
+      return a.isVisualImpaired < b.isVisualImpaired;
+    }
+    if (a.bitrate != b.bitrate)
+    {
+      return a.bitrate < b.bitrate;
+    }
+    return a.samplerate < b.samplerate;
+  }
+};
+
+struct SortComparerStreamSubtitle
+{
+  bool operator()(const SubtitleStreamInfoExt& a, const SubtitleStreamInfoExt& b)
+  {
+    if (a.languageDesc != b.languageDesc)
+    {
+      return a.languageDesc < b.languageDesc;
+    }
+    if (a.isDefault != b.isDefault)
+    {
+      return a.isDefault < b.isDefault;
+    }
+    if (a.isForced != b.isForced)
+    {
+      return a.isForced < b.isForced;
+    }
+    if (a.isHearingImpaired != b.isHearingImpaired)
+    {
+      return a.isHearingImpaired < b.isHearingImpaired;
+    }
+    if (a.isVisualImpaired != b.isVisualImpaired)
+    {
+      return a.isVisualImpaired < b.isVisualImpaired;
+    }
+    return a.codecName < b.codecName;
+  }
+};
+
+} // unnamed namespace
+
+void KODI::VIDEO::GUILIB::OpenDialogSelectVideoStream()
+{
+  CGUIDialogSelect* dialog{CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogSelect>(
+      WINDOW_DIALOG_SELECT_VIDEO_STREAM)};
+  if (!dialog)
+  {
+    CLog::LogF(LOGERROR, "Unable to get WINDOW_DIALOG_SELECT_VIDEO_STREAM dialog instance");
+    return;
+  }
+
+  auto& components = CServiceBroker::GetAppComponents();
+  auto appPlayer = components.GetComponent<CApplicationPlayer>();
```

Comment:
Do we need to check for !appPlayer, or is appPlayer always assumed in the context of video code?

Reply:
by looking at `components.GetComponent<CApplicationPlayer>()` uses in the source code is never used check the pointer